File: app/src/main/java/com/ocram/qichwadic/domain/repository/FavoriteRepository.java
package com.ocram.qichwadic.domain.repository;

import com.ocram.qichwadic.domain.model.Favorite;

import java.util.List;

import io.reactivex.Flowable;

public interface FavoriteRepository {

    void addFavorite(Favorite favorite);

    void removeFavorite(Favorite favorite);

    void clearFavorites();

    Flowable<List<Favorite>> getFavorites();

}

File: app/src/main/java/com/ocram/qichwadic/features/favorites/domain/FavoriteRepository.kt
package com.ocram.qichwadic.features.favorites.domain

import com.ocram.qichwadic.core.domain.model.DefinitionModel
import kotlinx.coroutines.flow.Flow

interface FavoriteRepository {

    suspend fun getFavorites(page: Int): Flow<List<DefinitionModel>>

    suspend fun addFavorites(favorite: DefinitionModel): List<Long>

    suspend fun removeFavorite(favorite: DefinitionModel): Int

    suspend fun clearFavorites(): Int

}